src/vmm/src/devices/virtio/vhost_user_block/device.rs

```diff
@@ -303,3 +332,443 @@ impl VirtioDevice for VhostUserBlock {
+                _fd: &EventFd,
+            ) -> Result<(), vhost::Error> {
+                Ok(())
+            }
+
+            fn set_vring_kick(
+                &self,
+                _queue_index: usize,
+                _fd: &EventFd,
+            ) -> Result<(), vhost::Error> {
+                Ok(())
+            }
+
+            fn set_vring_enable(
+                &mut self,
+                _queue_index: usize,
+                _enable: bool,
+            ) -> Result<(), vhost::Error> {
+                unsafe { (*self.vring_enabled.get()) = true };
+                Ok(())
+            }
+        }
+
+        // Block creation
+        let tmp_dir = utils::tempdir::TempDir::new().unwrap();
+        let tmp_dir_path_str = tmp_dir.as_path().to_str().unwrap();
+        let tmp_socket_path = format!("{tmp_dir_path_str}/tmp_socket");
+
+        unsafe {
+            let socketfd = libc::socket(libc::AF_UNIX, libc::SOCK_STREAM, 0);
+            if socketfd < 0 {
+                panic!("Cannot create socket");
+            }
+            let mut socket_addr = libc::sockaddr_un {
+                sun_family: libc::AF_UNIX as u16,
+                sun_path: [0; 108],
+            };
+
+            std::ptr::copy::<i8>(
+                tmp_socket_path.as_ptr().cast(),
+                socket_addr.sun_path.as_mut_ptr(),
+                tmp_socket_path.as_bytes().len(),
+            );
+
+            let bind = libc::bind(
+                socketfd,
+                (&socket_addr as *const libc::sockaddr_un).cast(),
+                std::mem::size_of::<libc::sockaddr_un>() as u32,
+            );
+            if bind < 0 {
+                panic!("Cannot bind socket");
+            }
+
+            let listen = libc::listen(socketfd, 1);
+            if listen < 0 {
+                panic!("Cannot listen on socket");
+            }
+        }
+
+        let vhost_block_config = VhostUserBlockConfig {
+            drive_id: "test_drive".to_string(),
+            partuuid: None,
+            is_root_device: false,
+            cache_type: CacheType::Writeback,
+            socket: tmp_socket_path,
+        };
+        let mut vhost_block = VhostUserBlockImpl::<MockMaster>::new(vhost_block_config).unwrap();
+
+        // Memory creation
+        let region_size = 0x10000;
+        let file = TempFile::new().unwrap().into_file();
+        file.set_len(region_size as u64).unwrap();
+        let regions = vec![(
+            FileOffset::new(file.try_clone().unwrap(), 0x0),
+            GuestAddress(0x0),
+            region_size,
+        )];
+        let guest_memory = GuestMemoryMmap::from_raw_regions_file(regions, false, false).unwrap();
+
+        vhost_block.activate(guest_memory).unwrap();
+        assert!(unsafe { *vhost_block.vu_handle.vu.features_are_set.get() });
+        assert!(unsafe { *vhost_block.vu_handle.vu.memory_is_set.get() });
+        assert!(unsafe { *vhost_block.vu_handle.vu.vring_enabled.get() });
+    }
+}
```

Comment:
`VhostUserBlockImpl`'s `fn config()` and `fn read_config()` don't seem to be covered.

Reply:
added tests at the end of `test_new_all_features` test.